gz-grant.html

```diff
@@ -119,118 +134,178 @@
         attached: function() {
           console.log('gz-grant widget attached!')
         },
+
         grant: function() {
```

Comment:
I'm having a lot of trouble keeping the public variables synced across several components. If I manually assign all the variables, by the time I call `grant()`, the variables might have changed. 

I've tried changing the API to pass the values as input parameters and that's proving to be more reliable. Can you see a downside to it?

        grant: function(_grantee, _resource, _readonly) {
          const data = { grantee: _grantee,
                         resource: _resource,
                         readOnly: _readonly }